### PR TITLE
Work queues improvements

### PR DESCRIPTION
**Pull request by JindrichSusen**

### CODE REVIEW

File: backend/Origam.sln [hunk 1]
```diff
@@ -135,7 +135,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Workflow", "Workflow", "{00
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Gui", "Gui", "{B842C91A-E5F8-4377-BAE3-BAED36417630}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Origam.workflow.mail", "Origam.workflow.mail\Origam.workflow.mail.csproj", "{5E18CE83-B56F-44D3-A106-99CDAAB33C8B}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Origam.workflow.mail", "Origam.workflow.mail\Origam.workflow.mail.csproj", "{5E18CE83-B56F-44D3-A106-99CDAAB33C8B}"
```
Comment (washibana) [inline backend/Origam.sln:138]: It's suspicious that this project id is the same as the one on row 184.

File: test/startTests.sh [hunk 2]
```diff
@@ -20,3 +20,34 @@ fi
...
 yarn install --ignore-engines
 yarn test:e2e
+if [ $? -eq 0 ]
+then
+  echo "Success."
+else
+  echo "Scripts failed" >&2
+  exit 1
+fi
+echo "Start integration tests.";
```
Comment (washibana) [inline test/startTests.sh:30]: Please change the message to `Start workflow integration tests.`.

**Review by washibana — CHANGES_REQUESTED** [review]
Test results contain exceptions related to the runtime disconnection. Please review them.

```
Running DisconnectRuntime. There might be some errors logged here. These are probably not a problem.
server_1       | 
server_1       | 2023-04-24 12:57:28,468 [12] ERROR Origam.OrigamEngine.OrigamEngine - Could not get restart status. Will retry in 10seconds.
server_1       | System.NullReferenceException: Object reference not set to an instance of an object.
server_1       |    at Origam.OrigamEngine.OrigamEngine.GetLastRestartRequestDate() in D:\a\origam\origam\backend\Origam.OrigamEngine\OrigamEngine.cs:line 221
server_1       |    at Origam.OrigamEngine.OrigamEngine.IsRestartPending() in D:\a\origam\origam\backend\Origam.OrigamEngine\OrigamEngine.cs:line 230
server_1       |    at Origam.OrigamEngine.OrigamEngine.RestartTimer_Elapsed(Object sender, ElapsedEventArgs e) in D:\a\origam\origam\backend\Origam.OrigamEngine\OrigamEngine.cs:line 254
```

**Comment by JindrichSusen** [comment]
> 

This will be solved as a separate issue
https://community.origam.com/t/eliminate-exceptions-caused-by-runtimeservicefactory-unloadservices/2798

File: backend/Origam.sln (hunk 1 above)
Comment (JindrichSusen) [reply backend/Origam.sln:138]: The project ID "5E18CE83-B56F-44D3-A106-99CDAAB33C8B" is not the same as on the line 184. The project type id "9A19103F-16F7-4668-BE54-9A1E7A4F7556" is. Which is fine I think.

**Pull request merged**